dwitter/views_v2.py

```diff
@@ -70,18 +71,30 @@ class DweetViewSet(mixins.RetrieveModelMixin,
     queryset = Dweet.objects.all().select_related(
         'author',
     ).prefetch_related(
+        Prefetch('reply_to', queryset=Dweet.objects.select_related('author')),
         'likes',
```

Comment:
Maybe this one can be removed now that we annotate the count directly?

Reply:
Did this comment land the wrong place?

Referring to `annotate(awesome_count=Count('likes'))`?  
I'll play around to see if it can get removed.

Reply:
Actually, were talking about the conditional annotate (line 131 in the old file)? In any case that one can be removed now :)

Reply:
I thought maybe we could remove fetching every single like, if we only used it for getting the count.

Reply:
Oh yeah, you're right!
Down to 6 now:

1. django session
2. fetch user (from "rest_framework/views.py", in `perform_authentication`)
3. Subquery to fetch the dweets (3 joins)
4. Final dweet fetch (6 joins)
5. Fetch all the comments
6. Fetch all the remixes